.gitignore

```diff
@@ -69,6 +69,8 @@ target/
 .DS_Store
 Thumbs.db
 .ccm/
+checkstyle_cachefile
+checkstyle_test_cachefile
```

Comment:
could we just put these into the build directory instead ? or the user's tmp directory? (i don't know how it works across different clones and repos…?)

(otherwise this will break releases)

Reply:
I moved it into the build dir. If it were on the tmp folder there could be cross-talk between different builds.

Reply:
Devbranch job seems to have been able to build ok...

Reply:
@bereng @michaelsembwever  Could we make it a property? That way I can put whatever path I want to `build.properties` file. Once it is in `build` dir, it will be there forever no?

What if I do `ant realclean jar`? Even I have not changed single line of the source code, I will need to do checkstyle again?

People may have a global '.gitignore' in their $HOME (normal git stuff) so I can point my build to cache it somewhere it will be gitignored anyway.

Reply:
> Even I have not changed single line of the source code, I will need to do checkstyle again?

You compile classes again, even if you haven't touched them. That's the point with clean, and the behaviour i would expect.

I'm weary of having anything at the top level, it causes problems.  Having it under build/ (specifically `${build.dir}`) in in line with the work happening in 18133.

I would also be in favour of not calling checkstyle when jar building. With 18133, all linters are a separate command (e.g. .build/check-code.sh).  To me this makes sense, has we also have owasp which is critical but we are ignoring because it's too slow (and network prone).

Reply:
@bereng  yes, `build.properties` is not there but `build.propeties.default` is. That is supposed to be a template you place your properties and and rename it to `build.properties` which is in our `.gitignore`.

Anyway, I gave it the second thought and I think you all are right. `realclean` should just clean it _all_. We definitely need to have a target which wipes it all out regardless of the properties / settings a developer might set.